File: app/src/main/java/com/mxt/anitrend/data/converter/UserStatsConverter.java
package com.mxt.anitrend.data.converter;

import com.mxt.anitrend.model.api.retro.WebFactory;
import com.mxt.anitrend.model.entity.general.UserStats;

import io.objectbox.converter.PropertyConverter;

/**
 * Created by max on 2017/11/04.
 * Entity Converter
 */

public class UserStatsConverter implements PropertyConverter<UserStats, String> {

    @Override
    public synchronized UserStats convertToEntityProperty(String databaseValue) {
        if(databaseValue == null)
            return null;
        return WebFactory.gson.fromJson(databaseValue, UserStats.class);
    }

    @Override
    public synchronized String convertToDatabaseValue(UserStats entityProperty) {
        if(entityProperty == null)
            return null;
        return WebFactory.gson.toJson(entityProperty);
    }
}

File: app/src/main/java/com/mxt/anitrend/data/converter/UserStatsConverter.kt
package com.mxt.anitrend.data.converter

import com.mxt.anitrend.model.api.retro.WebFactory
import com.mxt.anitrend.model.entity.anilist.UserStats
import io.objectbox.converter.PropertyConverter

/**
 * Created by max on 2017/11/04.
 * Entity Converter
 */
@Suppress("DEPRECATION")
class UserStatsConverter : PropertyConverter<UserStats, String> {
    @Synchronized
    override fun convertToEntityProperty(databaseValue: String?): UserStats? = databaseValue?.let { WebFactory.gson.fromJson(it, UserStats::class.java) }

    @Synchronized
    override fun convertToDatabaseValue(entityProperty: UserStats?): String? = entityProperty?.let { WebFactory.gson.toJson(it) }
}
